tests/lib/RequestParser.spec.js

```diff
@@ -0,0 +1,221 @@
+            error = e;
+        }
+        expect(error).toEqual(new Errors.InvalidRequestError('keyId must be of type string'));
+
+
+        keyId = '2ec615522906'; // dummy data
+        let parsedKeyInfo = await requestParser.parseKeyId(keyId);
+        expect(new KeyInfo(
+            '2ec615522906',
+            Key.Type.LEGACY,
+            true,
+            false,
+        )).toEqual(parsedKeyInfo);
+    });
+
+    it('can parse indiceArrays', () => {
+        const requestParser = new RequestParser();
+
+        /** @type {any} */
+        let indicesArray = undefined;
+        expect( () => { requestParser.parseIndicesArray(indicesArray); }).toThrow();
+        indicesArray = { x: 7 };
+        expect( () => { requestParser.parseIndicesArray(indicesArray); }).toThrow();
+        indicesArray = 'SindDasIndices';
+        expect( () => { requestParser.parseIndicesArray(indicesArray); }).toThrow();
+        indicesArray = 9;
+        expect( () => { requestParser.parseIndicesArray(indicesArray); }).toThrow();
+        indicesArray = [];
+        expect( () => { requestParser.parseIndicesArray(indicesArray); }).toThrow();
+        indicesArray = [1,2,3,4,5];
+        expect( () => { requestParser.parseIndicesArray(indicesArray); }).toThrow();
+        indicesArray = ["1","2","3"];
+        expect( () => { requestParser.parseIndicesArray(indicesArray); }).toThrow();
+        indicesArray = ["1'","2'","3'"];
+        expect(requestParser.parseIndicesArray(indicesArray)).toEqual(indicesArray);
+    });
+
+    it('can parse transaction', () => {
+        const requestParser = new RequestParser();
+
+        /** @type {any} */
+        let transaction = undefined;
+        expect( () => { requestParser.parseTransaction(transaction); }).toThrow();
+        transaction = 5;
+        expect( () => { requestParser.parseTransaction(transaction); }).toThrow();
+        transaction = {};
+        expect( () => { requestParser.parseTransaction(transaction); }).toThrow();
+
+        transaction = {
+            data: new Uint8Array([84, 104, 97, 110, 107, 32, 121, 111, 117, 32, 102, 111, 114, 32, 115, 104, 111, 112, 112, 105, 110, 103, 32, 97, 116, 32, 115, 104, 111, 112, 46, 110, 105, 109, 105, 113, 46, 99, 111, 109, 32, 40, 72, 51, 88, 67, 48, 68, 41]),
+            fee: 0,
+            recipient: new Uint8Array([225, 253, 0, 255, 238, 105, 158, 173, 122, 16, 27, 203, 31, 16, 3, 178, 231, 105, 81, 188]),
+            sender: new Uint8Array([238, 61, 13, 183, 158, 200, 247, 106, 130, 61, 9, 123, 134, 82, 60, 95, 16, 71, 39, 70]),
+            senderType: 0,
+            validityStartHeight: 176450,
+            value: 545000000,
+        };
+        expect(requestParser.parseTransaction(transaction)).toEqual(
+            new Nimiq.ExtendedTransaction(
+                new Nimiq.Address(new Uint8Array([238, 61, 13, 183, 158, 200, 247, 106, 130, 61, 9, 123, 134, 82, 60, 95, 16, 71, 39, 70])), //sender
+                0, // senderType
```

Comment:
Maybe use the Nimiq type directly? Otherwise we would kind of be testing the Nimiq lib, too. What do you think?
```suggestion
                Nimiq.Account.BASIC, // senderType
```